File: app/src/test/java/com/yeonsik/fitnessapp/data/MassUnitTest.java
package com.yeonsik.fitnessapp.data;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public final class MassUnitTest {
    @Test
    public void usesTheExactPoundToKilogramConversion() {
        assertEquals(0.45359237d, MassUnit.KILOGRAMS_PER_POUND, 0d);
        assertEquals(102.05828325d, MassUnit.toKg(225d, MassUnit.LB), 1e-12d);
        assertEquals(225d, MassUnit.fromKg(102.05828325d, MassUnit.LB), 1e-12d);
    }

    @Test
    public void convertsAndRoundTripsWithoutChangingThePhysicalLoad() {
        double pounds = 187.5d;
        double kilograms = MassUnit.toKg(pounds, MassUnit.LB);
        assertEquals(kilograms, MassUnit.convert(pounds, MassUnit.LB, MassUnit.KG), 0d);
        assertEquals(pounds, MassUnit.convert(kilograms, MassUnit.KG, MassUnit.LB), 1e-12d);
    }
}

File: shared/src/commonTest/kotlin/com/yeonsik/fitness/shared/feature/workout/model/MassUnitTest.kt
package com.yeonsik.fitness.shared.feature.workout.model

import kotlin.test.Test
import kotlin.test.assertEquals

class MassUnitTest {
    @Test
    fun usesTheExactPoundToKilogramConversion() {
        assertEquals(0.45359237, MassUnit.KILOGRAMS_PER_POUND, 0.0)
        assertEquals(102.05828325, MassUnit.toKg(225.0, MassUnit.LB), 1e-12)
        assertEquals(225.0, MassUnit.fromKg(102.05828325, MassUnit.LB), 1e-12)
    }

    @Test
    fun convertsAndRoundTripsWithoutChangingThePhysicalLoad() {
        val pounds = 187.5
        val kilograms = MassUnit.toKg(pounds, MassUnit.LB)

        assertEquals(kilograms, MassUnit.convert(pounds, MassUnit.LB, MassUnit.KG), 0.0)
        assertEquals(pounds, MassUnit.convert(kilograms, MassUnit.KG, MassUnit.LB), 1e-12)
    }
}
